src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs

```diff
@@ -486,5 +477,21 @@ internal IAsyncResult BaseBeginWrite(byte[] buffer, int offset, int count, Async
             => base.BeginWrite(buffer, offset, count, callback, state);
 
         internal void BaseEndWrite(IAsyncResult asyncResult) => base.EndWrite(asyncResult);
+
+        internal static bool IsIoRelatedException(Exception e) =>
+            // These all derive from IOException
+            //     DirectoryNotFoundException
+            //     DriveNotFoundException
+            //     EndOfStreamException
+            //     FileLoadException
+            //     FileNotFoundException
+            //     PathTooLongException
+            //     PipeException
+            e is IOException ||
+            // Note that SecurityException is only thrown on runtimes that support CAS
+            // e is SecurityException ||
+            e is UnauthorizedAccessException ||
+            e is NotSupportedException ||
+            (e is ArgumentException && !(e is ArgumentNullException));
```

Comment:
How was this list arrived at?

Reply:
In my previous PR I've just moved it from `FileStream.cs` to `LegacyFileStreamStrategy.cs`:

https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/System.Private.CoreLib/src/System/IO/LegacyFileStreamStrategy.cs#L386-L400

It comes from `FileStream` (5.0 branch):

https://github.com/dotnet/runtime/blob/832a65623d180412fcbaefd68aa57065341db931/src/libraries/System.Private.CoreLib/src/System/IO/FileStream.cs#L710-L724